bin/create-app-secret

```diff
@@ -1,17 +1,57 @@
 #!/bin/bash
-read -s -p "Enter the Discovery Django secret key: " discovery_django_secret_key
-echo
-if [ -z "${discovery_django_secret_key}" ]; then
-  echo "The Discovery Django secret key cannot be empty."
-  exit 1
-fi
-read -s -p "Re-enter the secret key: " discovery_django_secret_key2
-echo
-if [ ! "${discovery_django_secret_key}" = "${discovery_django_secret_key2}" ]; then
-  echo "Discovery Django secret keys do not match."
-  exit 1
-fi
 
-podman secret rm discovery-django-secret-key >/dev/null 2>&1
-printf "${discovery_django_secret_key}" | podman secret create discovery-django-secret-key -
-podman secret ls --filter name=discovery-django-secret-key
+function set_django_secret_key {
+  podman secret rm discovery-django-secret-key >/dev/null 2>&1
+  printf '%s' "$1" | podman secret create discovery-django-secret-key -
+  podman secret ls --filter name=discovery-django-secret-key
+}
+
+function generate_key {
+  local discovery_django_secret_key
+  discovery_django_secret_key=$(head -c 64 /dev/urandom | base64 | tr -dc 'a-zA-Z0-9')
+  set_django_secret_key "${discovery_django_secret_key}"
+}
+
+function prompt_key {
+  local discovery_django_secret_key
+  read -srp "Enter the Discovery Django secret key: " discovery_django_secret_key
+  echo
+  if [ -z "${discovery_django_secret_key}" ]; then
+    echo "The Discovery Django secret key cannot be empty."
+    exit 1
+  fi
+  read -srp "Re-enter the secret key: " discovery_django_secret_key2
+  echo
+  if [ ! "${discovery_django_secret_key}" = "${discovery_django_secret_key2}" ]; then
+    echo "Discovery Django secret keys do not match."
+    exit 1
+  fi
+  set_django_secret_key "${discovery_django_secret_key}"
+}
+
+function prompt_for_generation {
+  while true; do
+    read -rp "Automatically generate new Django secret key? [y/n]: " yn
```

Comment:
will need minor update to README.md and https://gitlab.cee.redhat.com/discovery/unofficial-internal-documentation/-/blob/main/how-to-guides/Deployments/Quadlet.md

Reply:
https://gitlab.cee.redhat.com/discovery/unofficial-internal-documentation/-/merge_requests/52